lib/instream/query/builder.ex

```diff
@@ -131,6 +131,9 @@ defmodule Instream.Query.Builder do
   Builds a `WHERE` query expression.
   """
   @spec where(t, map) :: t
+  def where(query, nil   ), do: query
+  def where(query, []    ), do: query
+  def where(query, fields) when fields == %{}, do: query
```

Comment:
I am unsure if this is the right place for this change. When only looking at some code I would expect the `where` function to always modify that part of the query. For example consider this situation:

```elixir
query_limited = BuilderSeries |> Builder.from() |> Builder.where(%{ answer 42 })
query_unlimited = query_limited |> Builder.where(nil)
```

What would one expect the second query does? With your change it is the same, as passing an empty-ish value is a no-op. Before it would clear the currently defined `where` parameters of the query.

I think this behavior should be handled by `Instream.Encoder.InfluxQL`. Currently there is support for `nil` (that is, no `WHERE` clause) and filled maps. The above example would remove the `WHERE` statement from the query if it would have been set before. Empty maps are broken at the moment and definitely need a fix (to behave like having `nil` set).

Apart from that I think the contract should stick with maps. Allowing a `Keyword.t` here would break later in the encoder (`append_where/2` calls `Map.keys/1` explicitly). Having the possibility of duplicate keys could also create queries that would never return a result:

```elixir
BuilderSeries
|> Builder.from()
|> Builder.where([ answer: 42, answer: 24 ])
|> InfluxQL.encode()

# "SELECT * FROM some_measurement WHERE answer=42 AND answer=24
```

The current `QueryBuilder` is completely oblivious of anything else than `AND` combinations. At least I find the time to attack the macro rewrite of that (see #18 for reference).